contracts/truefi2/Liquidator2.sol

```diff
@@ -129,31 +141,43 @@ contract Liquidator2 is UpgradeableClaimable {
     /**
      * @dev Liquidates a defaulted Loan, withdraws a portion of tru from staking pool
      * then transfers tru to TrueFiPool as compensation
-     * @param loan Loan to be liquidated
+     * @param loans Loan to be liquidated
      */
-    function liquidate(IDebtToken loan) external {
+    function liquidate(IDebtToken[] memory loans) external {
         require(msg.sender == SAFU, "Liquidator: Only SAFU contract can liquidate a loan");
-        require(loanFactory.isCreatedByFactory(address(loan)), "Liquidator: Unknown loan");
-        require(loan.status() == IDebtToken.Status.Defaulted, "Liquidator: Loan must be defaulted");
-        ITrueFiPool2 pool = ITrueFiPool2(loan.pool());
-        require(poolFactory.isSupportedPool(pool), "Liquidator: Pool not supported for default protection");
-        uint256 defaultedValue = loan.debt().sub(loan.repaid());
-        uint256 withdrawnTru = getAmountToWithdraw(defaultedValue, ITrueFiPoolOracle(pool.oracle()));
+        uint256 totalDefaultedValue;
+
+        for (uint256 i = 0; i < loans.length; i++) {
+            IDebtToken loan = loans[i];
+            require(loanFactory.isCreatedByFactory(address(loan)), "Liquidator: Unknown loan");
+            require(loan.status() == IDebtToken.Status.Defaulted, "Liquidator: Loan must be defaulted");
+            ITrueFiPool2 pool = ITrueFiPool2(loan.pool());
+            require(poolFactory.isSupportedPool(pool), "Liquidator: Pool not supported for default protection");
+
+            uint256 loanDefaultedValue = loan.debt().sub(loan.repaid());
+            totalDefaultedValue = totalDefaultedValue.add(getDefaultedValueInUsd(loanDefaultedValue, pool.oracle()));
+            loan.liquidate();
+        }
+
+        uint256 withdrawnTru = getAmountToWithdraw(totalDefaultedValue);
         stkTru.withdraw(withdrawnTru);
-        loan.liquidate();
         tru.safeTransfer(SAFU, withdrawnTru);
-        emit Liquidated(loan, defaultedValue, withdrawnTru);
+        emit Liquidated(loans, totalDefaultedValue, withdrawnTru);
     }
 
     /**
      * @dev Calculate amount of tru to be withdrawn from staking pool (not more than preset share)
      * @param deficit Amount of tusd lost on defaulted loan
      * @return amount of TRU to be withdrawn on liquidation
      */
-    function getAmountToWithdraw(uint256 deficit, ITrueFiPoolOracle oracle) internal view returns (uint256) {
+    function getAmountToWithdraw(uint256 deficit) internal view returns (uint256) {
         uint256 stakingPoolSupply = stkTru.stakeSupply();
         uint256 maxWithdrawValue = stakingPoolSupply.mul(fetchMaxShare).div(BASIS_RATIO);
-        uint256 deficitInTru = oracle.tokenToTru(deficit);
+        uint256 deficitInTru = tusdPoolOracle.tokenToTru(deficit);
```

Comment:
Let's add a comment explaining why we're favoring the TUSD pool (TUSD is already 18 decimals, so this would only work for 18 decimal precision on USD stablecoins).